apps/api/src/view-models/buildViewModelForResource/viewModels/interfaces/ICategoryTree.ts

```diff
@@ -0,0 +1,12 @@
+import { ResourceOrNoteCompositeIdentifier } from '../../../../domain/models/categories/types/ResourceOrNoteCompositeIdentifier';
+import { ViewModelId } from '../types/ViewModelId';
+
+export interface ICategoryTree {
```

Comment:
This interface is only implemented once. I'd remove it, unless the separate interface is necessary to get the recursive type. Can you have the children property of the View Model class typed as an array of the same class?

Also, it may pay to have a more general utility type for trees, the next time we have to deal with a tree type.

Reply:
Removed the interface.